### PR TITLE
iio: single parameter iio api for pluto

### PR DESCRIPTION
**Pull request by mormj**
This starts with the pluto source (and the inherited blocks it affects) and as of yet only has a few methods that are implemented. Need to tie the dependent parameters together

rather than complex constructors and a catch all set_params, construct
with the minimum amount of parameters and then have them individually
settable

### CODE REVIEW

**Comment by MountainLogic** [comment]
@adamhorden ,  Thanks so much for your work on iio.  Its a big deal for me.  I have pulled this branch of gnuradio and it is building fine.  I am seeing qa_iio fail:

`scott@scott:~/wip/gnuradio/build$ ctest -R qa_iio -V
UpdateCTestConfiguration  from :/home/scott/wip/gnuradio/build/DartConfiguration.tcl
UpdateCTestConfiguration  from :/home/scott/wip/gnuradio/build/DartConfiguration.tcl
Test project /home/scott/wip/gnuradio/build
Constructing a list of tests
Done constructing a list of tests
Updating test list for fixtures
Added 0 tests to meet fixture requirements
Checking test dependency graph...
Checking test dependency graph end
test 222
    Start 222: qa_iio

222: Test command: /usr/bin/sh "/home/scott/wip/gnuradio/build/gr-iio/python/iio/qa_iio_test.sh"
222: Test timeout computed to be: 10000000
222: Traceback (most recent call last):
222:   File "/home/scott/wip/gnuradio/gr-iio/python/iio/__init__.py", line 31, in <module>
222:     from .iio_python import *
222: ModuleNotFoundError: No module named 'gnuradio.iio.iio_python'
222: 
222: During handling of the above exception, another exception occurred:
222: 
222: Traceback (most recent call last):
222:   File "/home/scott/wip/gnuradio/gr-iio/python/iio/qa_iio.py", line 12, in <module>
222:     from gnuradio import gr, gr_unittest, iio, blocks
222:   File "/home/scott/wip/gnuradio/gr-iio/python/iio/__init__.py", line 35, in <module>
222:     from .iio_python import *
222: ImportError: generic_type: type "device_source" referenced unknown base type "gr::sync_block"
1/1 Test #222: qa_iio ...........................***Failed    0.29 sec

0% tests passed, 1 tests failed out of 1

Total Test time (real) =   0.32 sec

The following tests FAILED:
	222 - qa_iio (Failed)
Errors while running CTest
`

**Comment by adamhorden** [comment]
I am really liking this approach @mormj 🤓 . I know it is a work in progress, but looks a much cleaner interface 😁 .

**Adam Horden**

**Comment by tfcollins** [comment]
There is some logic missing here to correctly handle the filter modes: https://github.com/gnuradio/gnuradio/blob/a96427124579add8aa06e5cc1593a91947238fd7/gr-iio/lib/fmcomms2_sink_impl.cc

Likely resulting in something like "cannot set out_voltage_sample_rate ..." sometimes.

Let me try and pull a patch together this weekend.

-Travis

File: gr-iio/grc/iio.tree.yml [hunk 1]
```diff
@@ -7,11 +7,6 @@
     - iio_attr_sink
     - iio_attr_updater
     - iio_dds_control
-  - 'FMComms':
-    - iio_fmcomms2_source
-    - iio_fmcomms2_sink
-    - iio_fmcomms5_source
-    - iio_fmcomms5_sink
```
Comment (Yamakaja) [inline gr-iio/grc/iio.tree.yml:14]: Is there a reason why you're dropping `fmcomms[25]`?

Comment (mormj) [reply gr-iio/grc/iio.tree.yml:14]: Namely it's to get the pluto support out there with the updated API, then add these back later as people who have the means to test them can try it out.

**Comment by tfcollins** [comment]
> There is some logic missing here to correctly handle the filter modes: https://github.com/gnuradio/gnuradio/blob/a96427124579add8aa06e5cc1593a91947238fd7/gr-iio/lib/fmcomms2_sink_impl.cc
> 
> Likely resulting in something like "cannot set out_voltage_sample_rate ..." sometimes.
> 
> Let me try and pull a patch together this weekend.
> 
> -Travis

Just an FYI: https://github.com/mormj/gnuradio/pull/5

**Comment by adamhorden** [comment]
@mormj your feature branch requires a rebase 🤓. I will then merge it in.
I have rebased `feature/gr-iio` bringing in latest changes.

**Adam Horden**

**Comment by mormj** [comment]
I'll try and knock that out in the morning

**Comment by adamhorden** [comment]
@mormj I tested this locally with a Pluto before merging. Thanks for your work on this 🤓 .

**Adam Horden**

**Pull request merged**